cmd/cluster-init/main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"os"
+	"path/filepath"
+)
+
+type options struct {
+	clusterName  string
+	releaseRepo  string
+	buildFarmDir string
+
+	//flagutil.GitHubOptions TODO: this will come in later I think...lets ignore github stuff for now
+}
+
+func (o options) String() string {
+	return fmt.Sprintf("cluster-name: %s\nrelease-repo: %s\nbuild-farm-dir: %s",
+		o.clusterName,
+		o.releaseRepo,
+		o.buildFarmDir)
+}
+
+func parseOptions() options {
+	var o options
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.clusterName, "cluster-name", "", "The name of the new cluster.")
+	fs.StringVar(&o.releaseRepo, "release-repo", "", "Path to the root of the openshift/release repository.")
+	fs.StringVar(&o.buildFarmDir, "build-farm-dir", "", "The name of the new build farm directory.")
+	//o.AddFlags(fs)
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("cannot parse args: ", os.Args[1:])
+	}
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.clusterName == "" {
+		return fmt.Errorf("--cluster-name must be provided")
+	}
+	if o.releaseRepo == "" {
+		return fmt.Errorf("--release-repo must be provided")
+	}
+	if o.buildFarmDir == "" {
+		return fmt.Errorf("--build-farm-dir must be provided")
```

Comment:
Is this not deterministic from the name of the new cluster, under the release repo ?

Reply:
It could be, I thought allowing the option to be provided would make the tool more "future proof" but for now we can just derive it from the cluster name pretty easily.

Reply:
So in the `build03` cluster case we have the `buildFarmDir` of `03_cluster`. We _could_ deterministically generate this from the cluster name (strip of alpha chars, use the numeric as a prefix, append `_cluster`), but I am not sure that that is _better_ than just accepting it as a parameter to the tool. What do you think?

Reply:
Let's change the directory structure to be 1:1 with cluster names - there's nothing we gain by having `03_cluster` instead of `build03`

Reply:
Should we modify the existing clusters (`build01`, and `build02`) to use this also? or just do this going forward? I am a bit concerned with missing something if I do the former, but I could do an exhaustive full-text search and take care of it if that is desired (possibly in a new PR).